src/editor.c

```diff
@@ -1234,15 +1236,65 @@ editor_get_indent_prefs(GeanyEditor *editor)
 }
 
 
+static inline gint compare_line_indentation(ScintillaObject *sci, gint line1, gint line2)
+{
+	gint line1_width = sci_get_line_indentation(sci, line1);
+	gint line2_width = sci_get_line_indentation(sci, line2);
+	return (line1_width < line2_width);
```

Comment:
return type is int, but actually returns a boolean, it's quite odd.  And the name suggests it returns a strcmp() like value.

Though, splitting in this function is maybe over-splitting